File: python_env/src/calculator.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
简单计算器模块示例
"""


class Calculator:
    """
    实现基本数学运算的计算器类
    """

    def add(self, a, b):
        """
        计算两个数的和
        
        Args:
            a: 第一个数
            b: 第二个数
        
        Returns:
            两个数的和
        """
        return a + b

    def subtract(self, a, b):
        """
        计算两个数的差
        
        Args:
            a: 第一个数
            b: 第二个数
        
        Returns:
            两个数的差
        """
        return a - b

    def multiply(self, a, b):
        """
        计算两个数的乘积
        
        Args:
            a: 第一个数
            b: 第二个数
        
        Returns:
            两个数的乘积
        """
        return a * b

    def divide(self, a, b):
        """
        计算两个数的商
        
        Args:
            a: 第一个数
            b: 第二个数
        
        Returns:
            两个数的商
        
        Raises:
            ZeroDivisionError: 当除数为零时抛出
        """
        if b == 0:
            raise ZeroDivisionError("除数不能为零")
        return a / b

File: ruby_env/lib/calculator.rb
# Calculator class
#
# A calculator class that implements basic mathematical operations
class Calculator
  # Initialize method
  def initialize
    # No initialization needed at the moment
  end
  
  # Calculate the sum of two numbers
  # @param a [Numeric] First number
  # @param b [Numeric] Second number
  # @return [Numeric] Sum of the two numbers
  def add(a, b)
    a + b
  end
  
  # Calculate the difference between two numbers
  # @param a [Numeric] First number
  # @param b [Numeric] Second number
  # @return [Numeric] Difference between the two numbers
  def subtract(a, b)
    a - b
  end
  
  # Calculate the product of two numbers
  # @param a [Numeric] First number
  # @param b [Numeric] Second number
  # @return [Numeric] Product of the two numbers
  def multiply(a, b)
    a * b
  end
  
  # Calculate the quotient of two numbers
  # @param a [Numeric] First number
  # @param b [Numeric] Second number
  # @return [Float] Quotient of the two numbers (floating point result)
  # @raise [ZeroDivisionError] When the divisor is zero
  def divide(a, b)
    raise ZeroDivisionError, "Divisor cannot be zero" if b.zero?
    # Convert the result to a floating point number to ensure integer division also returns decimal results
    a.to_f / b
  end
end
